### PR TITLE
V10 compatibility

### LINKED ISSUE #260: Raise compatible TYPO3 version in composer

### LINKED ISSUE #262: ConfigurationManager constructor needs arguments
\EBT\ExtensionBuilder\Configuration\ExtensionBuilderConfigurationManager extends TYPO3\CMS\Extbase\Configuration\ConfigurationManager which needs   
@param \TYPO3\CMS\Extbase\Object\ObjectManagerInterface $objectManager
and
@param \TYPO3\CMS\Extbase\Service\EnvironmentService $environmentService
as constructor arguments in v10.x

I think we should not inherit from that class anymore but get configuration from a new instance of ConfigurationManager

### LINKED ISSUE #276: TYPO3 10 compatible?
Are there any plans to make this extension Typo3 10 compatible?I find this extension very useful for people who are starting developing typo3 extensions and are not familiar with the structure.

Thanks.

### PR DESCRIPTION
**Pull request by nicodh**
We should make all changes compatible to both versions 9.5 & 10.x
resolves #262 
resolves #260 
resolves #276

## TODOs

- [x]  https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.0/Breaking-87623-ReplaceConfigpersistenceclassesTyposcriptConfiguration.html :arrow_right: 
  - `Resources/Private/CodeTemplates/Extbase/Partials/ext_typoscript_setup_mapping.txtt`
  - `Resources/Private/CodeTemplates/Extbase/ext_typoscript_setup.txtt`
- [x] https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.3/Deprecation-89463-SwitchableControllerActions.html :arrow_right: remove field in sidebar and `Resources/Private/CodeTemplates/Extbase/Configuration/Flexforms/flexform.xmlt`
- [x] https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.0/Deprecation-87550-UseControllerClassesWhenRegisteringPluginsmodules.html
- [x] https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.0/Breaking-87193-DeprecatedFunctionalityRemoved.html :arrow_right:  `The array key uploadfolder in extensions ext_emconf.php files is obsolete and ignored.`
- [x] Move `\TYPO3\CMS\Extbase\Utility\ExtensionUtility::registerPlugin` to `Configuration/TCA/Overrides/tt_content.php`

### CODE REVIEW

**Comment by pquerner** [comment]
https://github.com/FriendsOfTYPO3/extension_builder/blob/d3a5c245445c76bb9a106b48c8177c23708c540a/Classes/ViewHelpers/Be/ConfigurationViewHelper.php#L36

should be

`
$baseUrl = ExtensionManagementUtility::extPath('extension_builder');
`

**Comment by gordon81** [comment]
add this Pull request #266 to  TYPO3 v10 compatibility that is was pquerner say

**Comment by DavidBruchmann** [comment]
> https://github.com/FriendsOfTYPO3/extension_builder/blob/d3a5c245445c76bb9a106b48c8177c23708c540a/Classes/ViewHelpers/Be/ConfigurationViewHelper.php#L36
> 
> should be
> 
> `$baseUrl = ExtensionManagementUtility::extPath('extension_builder');`

@pquerner your proposition is not working, I have another PR only about that file.

**Comment by pquerner** [comment]
It worked for me when I posted this. Not sure what changed as I havent tried it since.

**Comment by DavidBruchmann** [comment]
As I pointed out in the comments below my PR it works for all the JS files, but the variable baseUrl in the Json of the BE-Site will include a local path which will be blocked by the browser.
I remarked it in the browser-console by a single file which didn't load, I think it was `extension_builder/Resources/Public/jsDomainModeling/wireit/images/loading.gif`

**Comment by pquerner** [comment]
Okay, glad its working now! :)

**Comment by DavidBruchmann** [comment]
Yeah, still needs being accepted and merged ;-)

File: Classes/Service/ClassBuilder.php [hunk 1]
```diff
@@ -584,7 +584,7 @@ protected function buildActionMethod(Model\DomainObject\Action $action, Model\Do
             $actionMethod->updateParamTags();
 
             if ($actionName === 'edit') {
-                $actionMethod->setTag('ignorevalidation', '$' . $parameterName);
+                $actionMethod->setTag('TYPO3\CMS\Extbase\Annotation\IgnoreValidation' . '("' . $parameterName . '")');
```
Comment (liayn) [inline Classes/Service/ClassBuilder.php:587]: And one more nitpick:
`'TYPO3\CMS\Extbase\Annotation\IgnoreValidation' . '("' . $parameterName . '")'`
should be
`'TYPO3\CMS\Extbase\Annotation\IgnoreValidation("' . $parameterName . '")'`

File: Classes/Configuration/ExtensionBuilderConfigurationManager.php [hunk 2]
```diff
@@ -110,13 +126,13 @@ public function getSubActionFromRequest()
...
-        if ($typoscript == null) {
-            $typoscript = $this->getConfiguration(parent::CONFIGURATION_TYPE_FULL_TYPOSCRIPT);
+        if ($typoscript === null) {
+            $typoscript = $this->configurationManager->getConfiguration($this->configurationManager::CONFIGURATION_TYPE_FULL_TYPOSCRIPT);
         }
         $settings = $typoscript['module.']['extension_builder.']['settings.'];
         $settings['extConf'] = $this->getExtensionBuilderSettings();
         if (empty($settings['publicResourcesPath'])) {
-            $settings['publicResourcesPath'] = ExtensionManagementUtility::siteRelPath('extension_builder') . 'Resources/Public/';
+            $settings['publicResourcesPath'] = ExtensionManagementUtility::extPath('extension_builder') . 'Resources/Public/';
```
Comment (simonschaufi) [inline Classes/Configuration/ExtensionBuilderConfigurationManager.php:135]: TODO

File: composer.json [hunk 3]
```diff
@@ -23,9 +23,9 @@
   ],
   "require": {
     "ext-json": "*",
-    "typo3/cms-core": "~9.5",
     "nikic/php-parser": "^4.0",
-    "typo3/cms-backend": "~9.5"
+    "typo3/cms-core": "^9.5 || ^10",
```
Comment (simonschaufi) [inline composer.json:27]: TODO: v10 only

**Comment by simonschaufi** [comment]
:warning: Please leave the branch in place as some people depend on it.

**Comment by liayn** [comment]
> 
> 
> ⚠️ Please leave the branch in place as some people depend on it.

@simonschaufi 
I would actually do the exact opposite. Remove it ASAP.
We will be flooded with outdated bug reports because people did not change to the master branch and will probably not be aware unless composer tells them that the branch is gone!

**Pull request merged**